Replace custom progress with cleaner tqdm

JohannLange opened the pull request:
[`tqdm`](https://github.com/tqdm/tqdm) is a very clean and fast way to display the progress of a loop. Just as the currently used custom progress prints, it displays the progress in percent as well as the elapsed and total number of timesteps, the elapsed and remaining time and the time per iteration.

Unless the plan was to display further information in the progress, I find `tqdm` to be way cleaner.
Also, I think all other relevant performance metrics are better logged into Tensorboard and not the shell.
But feel free to reject the PR if you had some more stuff planned for the progress display.

Toni-SM commented:
Hi @JohannLange,

It is a good idea to change the way progress is displayed... however, the pull request is a bit unclean.
Please, open a new pull request only with the minimum necessary modification for integrating `tqdm` as described below:

- modified files: `trainer/torch/base.py`, `trainer/torch/sequential.py`, `trainer/torch/paraller.py` and `setup.py`
- don't modify whitespace, blank lines, etc. Those cosmetic changes do not add anything substantial to the library's functionality.
- add import statements as follow:
  ```python
  function annotation
  <empty line>
  python libraries and other libraries (e.g. tqdm)
  <empty line>
  machine learning framework (e.g. torch)
  <empty line>
  skrl components
  ```
One question, have you tested the changes with the parallel trainer? Multiprocessing implementation may not be compatible with some modules containing anonymous functions or locks.

Btw, if you are interested in making further contributions, I have a list of future relevant implementations and pending tasks... it can be an interesting starting point :)

JohannLange commented:
Hi @Toni-SM,

thank you for your feedback!
Nevertheless, I have some questions/comments we should discuss before merging the PR.

> * modified files: `trainer/torch/base.py`, `trainer/torch/sequential.py`, `trainer/torch/paraller.py` and `setup.py`

First things first, sorry for re-adding the GitHub workflow, that was by mistake.
But regarding the examples in the docs, I updated them and removed the `"progress_interval"` setting, because it will be deprecated if you decide to merge the PR.

> * don't modify whitespace, blank lines, etc. Those cosmetic changes do not add anything substantial to the library's functionality.

Well here we both have a clearly different opinion.
For me removing trailing whitespaces mainly comes down to two reasons based on my comfort, besides most IDEs removing them anyway, first, [it makes navigating with keyboard commands unnecessarily hard](https://softwareengineering.stackexchange.com/questions/121555/why-is-trailing-whitespace-a-big-deal), and second, it creates a lot of "noise" with activated white space symbols like interpunct for spaces, right arrow for tabs and the [pilcrow](https://en.wikipedia.org/wiki/Pilcrow).

Also, while removing the trailing spaces doesn't add anything to skrl, neither does keeping them.

And lastly, [pep 8 recommends removing them anyway, best with a pre-commit hook](https://peps.python.org/pep-0008/#:~:text=Avoid%20trailing%20whitespace%20anywhere.%20Because%20it%E2%80%99s%20usually%20invisible%2C%20it%20can%20be%20confusing%3A%20e.g.%20a%20backslash%20followed%20by%20a%20space%20and%20a%20newline%20does%20not%20count%20as%20a%20line%20continuation%20marker.%20Some%20editors%20don%E2%80%99t%20preserve%20it%20and%20many%20projects%20(like%20CPython%20itself)%20have%20pre%2Dcommit%20hooks%20that%20reject%20it.), the later I could easily add if you wish to.

Oh and also that empty line at the end of file is really something that should be there, see [stackoverflow](https://stackoverflow.com/questions/2287967/why-is-it-recommended-to-have-empty-line-in-the-end-of-a-source-file) and [unix.stackexchange](https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file). Basically, it boils down to an old decision in [ANSI C 1989](https://gcc.gnu.org/legacy-ml/gcc/2003-11/msg01568.html) and was later incorporated into the [POSIX standard](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_206)

> * add import statements as follow:
>   ```python
>   function annotation
>   <empty line>
>   python libraries and other libraries (e.g. tqdm)
>   <empty line>
>   machine learning framework (e.g. torch)
>   <empty line>
>   skrl components
>   ```

That I can live with, I will reorder the import where I distorted the order and push a new commit.

> One question, have you tested the changes with the parallel trainer? Multiprocessing implementation may not be compatible with some modules containing anonymous functions or locks.

I tested it for [gym](https://skrl.readthedocs.io/en/latest/intro/examples.html#learning-in-a-gym-vectorized-environment-one-agent-multiple-environments) and [Isaac Gym](https://skrl.readthedocs.io/en/latest/intro/examples.html#learning-by-scopes-in-an-isaac-gym-environment-multiple-agents-and-environments) and it worked just fine.
Although I couldn't test it for Omniverse Gym/Sim because I don't use that, but I would suspect it behaves the same.
I hope that answers your question.

> Btw, if you are interested in making further contributions, I have a list of future relevant implementations and pending tasks... it can be an interesting starting point :)

Yes! I would happily help as I plan to further continue using skrl for its ease of configuration!

Toni-SM commented:
> Yes! I would happily help as I plan to further continue using skrl for its ease of configuration!

Glad to hear about it...
Here (https://github.com/users/Toni-SM/projects/2/views/8) is the pending tasks board in case you want to take a look
